test/unit/io/stream/parse_condition_test.cpp

```diff
@@ -0,0 +1,509 @@
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/io/stream/parse_condition.hpp>
+
+using namespace std::literals;
+using namespace seqan3;
+
+template <char char_v>
+struct foo : seqan3::detail::parse_condition<foo<char_v>>
+{
+    inline static constexpr constexpr_string msg{constexpr_string{"foo_"} + constexpr_string{char_v}};
+
+    using seqan3::detail::parse_condition<foo<char_v>>::parse_condition;
+
+    bool operator()(char c) const
+    {
+        return c == char_v;
+    }
+};
+
+struct bar
+{
+    template <typename alphabet_t>
+    bool operator()([[maybe_unused]] alphabet_t c) const
+    {
+        return true;
+    }
+};
+
+TEST(parse_condition, parse_condition)
+{
+    foo<'a'> p{};
+    EXPECT_TRUE(p('a'));
+    EXPECT_FALSE(p('f'));
+}
+
+TEST(parse_condition, parse_condition_msg)
+{
+    EXPECT_EQ(foo<'o'>::msg.string(), "foo_o"s);
+}
+
+TEST(parse_condition, parse_condition_concept)
+{
+    using namespace seqan3;
+
+    EXPECT_TRUE(detail::parse_condition_concept<is_in_alphabet<dna4>>);
+    EXPECT_TRUE(detail::parse_condition_concept<is_char<to_char(aa27::A)>>);
+    EXPECT_TRUE((detail::parse_condition_concept<is_in_interval<'a','z'>>));
+    EXPECT_TRUE(detail::parse_condition_concept<decltype(is_space)>);
+    EXPECT_TRUE(detail::parse_condition_concept<decltype(is_blank)>);
+    EXPECT_TRUE(detail::parse_condition_concept<decltype(is_graph)>);
+    EXPECT_TRUE(detail::parse_condition_concept<decltype(is_alpha)>);
+    EXPECT_TRUE(detail::parse_condition_concept<decltype(is_digit)>);
+    EXPECT_TRUE(detail::parse_condition_concept<decltype(is_alnum)>);
+    // TODO(rrahn): Direct call in concept caused ICE.
```

Comment:
has this issue been reported? Can you reference it here so that someone reading the comment can check if it has been fixed and remove the workaround?

Reply:
Not yet. I'll add it later. Is on my TODO.